codec/proto_codec.go

```diff
@@ -11,13 +11,21 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
+// ProtoCodecMarshaler defines and interface for codecs that utilize Protobuf for both
+// binary and JSON encoding.
+type ProtoCodecMarshaler interface {
```

Comment:
```suggestion
type ProtoMarshaler interface {
```

Reply:
The ProtoMarshaler type is already defined in that package.

Reply:
Maybe `ProtoMarshalerI` to align with other interface names in the SDK like `AccountI` ?

Reply:
This is a terrible name but I understand the reasoning. We should rename the other `ProtoMarshaler` separately

Reply:
Actually how about `ProtoCodecI`. Maybe that's best?